src/grid/grid.component.ts

```diff
@@ -318,12 +321,16 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         return this.rowList.toArray()[index];
     }
 
+    public getRowByKey(keyValue: any): IgxGridRowComponent {
+        return this.rowList.find((row) => row.primaryValue === keyValue);
```

Comment:
Is this needed _in addition_ to the API service method?

Reply:
I added it as a counterpart to the getRowByIndex method (as there is no other exposed method to get a row reference to a row). We could refactor getRowByIndex to incorporate the gridAPI get_row method and have it except both index or key?
```
public getRow(keyOrIndex: any): IgxGridRowComponent {
        return this.gridAPI.get_row(this.id, keyOrIndex);
    }
```